File: CodingInterviewsPackSolutionsCPP/Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int>myMap;
        int n = nums.size();
        for(int i = 0; i < n; i++){
            if(myMap.count(nums[i]) && i-myMap[nums[i]]<=k)
                return true;

            myMap[nums[i]] = i;
        }
        return false;
    }
};

File: CodingInterviewsJavaSolutions/Contains Duplicate II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> myMap = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (myMap.containsKey(nums[i]) && i - myMap.get(nums[i]) <= k) {
                return true;
            }
            myMap.put(nums[i], i);
        }
        return false;
    }
}
